DMCompiler/DM/Builders/DMExpressionBuilder.cs

```diff
@@ -716,7 +717,8 @@ private static DMExpression BuildDereference(DMASTDereference deref, DMObject dm
                             }
 
                             operations = new Dereference.Operation[newOperationCount];
-                            astOperationOffset = i + 1;
+                            astOperationOffset = i + iteration + 1;
+                            iteration++;
```

Comment:
A simpler solution would be `astOperationOffset += i + 1`
```suggestion
                            astOperationOffset += i + 1;
```

Reply:
Hold on, this appears to break /tg/ compile